### PR TITLE
Store last seen workspaces in local storage

### PR DESCRIPTION
**Pull request by MarcelGeo**
**Details**

Reaction to ticket https://github.com/MerginMaps/server-private/issues/2229

:hammer: 
 Store last seen workspace(s) in local storage instead of cookies. This PR is mostly only refactor, but following improvements are coming:
 
 - local storage is more robust than cookies
 - currentWorkspace cookies will not be sending in every request on API :1st_place_medal: 
 - there is not need to handle expiring of cookies
 - last workspace will be array instead key value (used for showing last seen workspace to user)
 
:heavy_plus_sign: New feature:

- We can now use Proxy based storing to local storage with ability to set value and sync it with localStorage. In case of array value in Proxy, use spread operator (push/pop ... are not reactive). This is experimental feature.

```javascript
const storage = createStorage('name-of-key', 0)

storage.value = 1
```

### CODE REVIEW

File: web-app/packages/lib/src/modules/user/store.ts [hunk 1]
```diff
@@ -380,167 +381,86 @@ export const useUserStore = defineStore('userModule', {
...
+    async cleanupLastSeenWorkspaces() {
+      this.lastWorkspaces.value = this.lastWorkspaces.value.filter(
+        (item) => item.userId !== this.loggedUser?.id
+      )
     },
 
+    /**
+     * Checks the current workspace that the user is in.
+     *
+     * Tries to get the current workspace ID from the cookies.
```
Comment (varmar05) [inline web-app/packages/lib/src/modules/user/store.ts:429]: we do not use cookies anymore, please update

File: web-app/packages/lib/src/modules/user/store.ts [hunk 2]
```diff
@@ -380,167 +381,86 @@ export const useUserStore = defineStore('userModule', {
...
-                // do not skip saving in cookies when using preferredWorkspace
-                skipSavingInCookies = false
-              }
+            const preferredWorkspaceId = this.loggedUser?.preferred_workspace
+            foundWorkspace = this.getWorkspaceById({
+              id: preferredWorkspaceId
+            })
+            if (foundWorkspace?.id) {
+              // do not skip saving in cookies when using preferredWorkspace
+              skipSavingInCookies = false
```
Comment (varmar05) [inline web-app/packages/lib/src/modules/user/store.ts:449]: do we need this?

**Review by varmar05 — COMMENTED** [review]
Nice one, looking good.

**Comment by MarcelGeo** [comment]
@varmar05 Just wondering, do we need this function in community edition with one workspace?

File: web-app/packages/lib/src/modules/user/store.ts (hunk 2 above)
Comment (MarcelGeo) [reply web-app/packages/lib/src/modules/user/store.ts:449]: I can rename variables . some workspace changes might not be persisted in storage due to specific logic.

**Comment by varmar05** [comment]
> @varmar05 Just wondering, do we need this function in community edition with one workspace?

We don't, if you can port it easily to EE/cloud, it would be good. Nevertheless backend APIs are available also for CE.

File: web-app/packages/lib/src/modules/user/store.ts (hunk 2 above)
Comment (varmar05) [reply web-app/packages/lib/src/modules/user/store.ts:449]: cool, just rename in that case

**Comment by MarcelGeo** [comment]
> > @varmar05 Just wondering, do we need this function in community edition with one workspace?
> 
> We don't, if you can port it easily to EE/cloud, it would be good. Nevertheless backend APIs are available also for CE.

We need to investigate potential solutions for porting specific store functions across different environments. It should be useful for error handling of specific cases in EE/Cloud. ```setWorkspace``` should be also ported with with option for storing of ws. I think this is not scope of this PR.

**Review by varmar05 — APPROVED** [review]

**Pull request merged**